src/Faithlife.Analyzers/IfNotNullFixAllProvider.cs

```diff
@@ -82,7 +83,7 @@ private IfNotNullFixAllProvider()
 					async (solutionAsync, document) =>
 						(await solutionAsync.ConfigureAwait(false))
 							.WithDocumentText(
-								document.Id,
+								document?.Id ?? throw new NullReferenceException(nameof(document)),
```

Comment:
Despite the null check on line 80 above, I was getting a `CS8602: (Dereference of a possibly null reference)` here.

Reply:
Using `WhereNotNull` from Faithlife.Utility would be better, IMO. If we don't want to introduce that dependency for some reason, you could use `document!.Id` because there is already a null check.

Reply:
```suggestion
								document!.Id,
```

The idiomatic way to express this in C# is the null-forgiving operator.

(An alternative would be updating the earlier code to move the null-forgiving closer to the test, i.e.,

```csharp
.Where(x => x is object) // is not null with <LangVersion>9.0</LangVersion>
.Select(x => x!)
```

but, as Dave says, it would be more straightforward to use `.WhereNotNull()`.)

Reply:
I've opted for `!` over `WhereNotNull` since Faithlife.Analyzers [is on netstandard1.3](https://github.com/Faithlife/FaithlifeAnalyzers/blob/4fbfbc97b892cb2c7eebecf08f1c74a81316689d/src/Faithlife.Analyzers/Faithlife.Analyzers.csproj#L5), which [isn't supported by Faithlife.Utility](https://github.com/Faithlife/FaithlifeUtility/blob/1edf1bd198c6f925df8f7c73cf15d24c86431172/src/Faithlife.Utility/Faithlife.Utility.csproj#L4) and I don't think I want to make a major version bump for that alone.